miteD/middleware/api.py

```diff
@@ -42,10 +70,52 @@ def stop(self):
                     with suppress(asyncio.CancelledError):
                         self._loop.run_until_complete(task)
                 self._loop.close()
+                self._deregister_with_consul()
 
             def get_remote_service(self, service_name, version):
                 return RemoteService(name=service_name, version=version, nc=self._nc)
 
+            def _register_with_consul(self):
+                self.consul_id = os.getenv("HOSTNAME")
+                # This is also the pod name
+
+                api_data = {
+                  "Name": self.consul_id,
+                  "ID": self.consul_id,
```

Comment:
the name and id being the same are presumably intentional?

Reply:
It is. In fact if we don't specify ID consul uses the name field but I wanted to be explicit. Happy to remove it if you think that's best.

Reply:
no I do not know enough of consul for that, just asking to ensure that it is intentional, no need to change